ObservatoryCore/LogMonitor.cs

```diff
@@ -347,11 +347,28 @@ private void HandleAncillaryFile(string eventType)
             // Brief sleep to ensure the content is updated before we read it.
             System.Threading.Thread.Sleep(50);
 
-            string fileContent = File.ReadAllText(journalWatcher.Path + Path.DirectorySeparatorChar + filename);
-
-            var fileObject = DeserializeToEventArgs(eventType + "File", fileContent);
-            JournalEntry?.Invoke(this, fileObject);
+            // Some files are still locked by another process after 50ms.
+            // Retry every 50ms for 0.5 seconds before giving up.
 
+            string fileContent = null;
+            int retryCount = 0;
+            
+            while (fileContent == null || retryCount < 10)
+            {
+                try
+                {
+                    using var fileStream = File.Open(journalWatcher.Path + Path.DirectorySeparatorChar + filename, FileMode.Open, FileAccess.Read, FileShare.ReadWrite);
+                    using var reader = new StreamReader(fileStream);
+                    fileContent = reader.ReadToEnd();
+                    var fileObject = DeserializeToEventArgs(eventType + "File", fileContent);
+                    JournalEntry?.Invoke(this, fileObject);
+                }
+                catch
+                {
+                    retryCount++;
+                    System.Threading.Thread.Sleep(50);
```

Comment:
Alternatively, put the sleep at the beginning of the while loop and delete the one before the loop.

6 of one, half dozen of the other.